keyboards/giabalanai/config.h

```diff
@@ -0,0 +1,226 @@
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0xFEED
+#define PRODUCT_ID   0x0000
+#define DEVICE_VER   0x0001
+#define MANUFACTURER 3araht
+#define PRODUCT      giabalanai
+
+/* key matrix size */
+#define MATRIX_ROWS 12
+#define MATRIX_COLS 10
+
+/*
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ */
+#define MATRIX_ROW_PINS { B5, B4, E6, D7, C6, D4 }
+#define MATRIX_COL_PINS { B1, F7, F6, F5, F4, B3, B2, B6, D0, D1 }
+#define MATRIX_ROW_PINS_RIGHT { D1, D0, D4, C6, D7, E6 }
+#define MATRIX_COL_PINS_RIGHT { F4, F5, F6, F7, B1, B3, B2, C7, B7, F1 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/*
+ * Split Keyboard specific options, make sure you have 'SPLIT_KEYBOARD = yes' in your rules.mk, and define SOFT_SERIAL_PIN.
+ */
+#define SOFT_SERIAL_PIN D2
+#define SELECT_SOFT_SERIAL_SPEED 1
+/*Sets the protocol speed when using serial communication*/
+//Speeds:
+//0: about 189kbps (Experimental only)
+//1: about 137kbps (default)
+//2: about 75kbps
+//3: about 39kbps
+//4: about 26kbps
+//5: about 20kbps
+
+// Right side has to be the master since 1, LED data is output from right side, and 2, Audio pin is prepared on right side as a reserve.
+#define MASTER_RIGHT
+
+
+//#define BACKLIGHT_PIN B7
+//#define BACKLIGHT_LEVELS 3
+//#define BACKLIGHT_BREATHING
+
+/* ws2812 RGB LED */
+#define RGB_DI_PIN D3
+
+#ifdef RGBLIGHT_ENABLE
+
+#    define RGBLED_NUM 123
+// Do not define "RGBLED_SPLIT" since the # of LEDs on both sides are not equal.
+// Instead, LED data is tranferred from right side to the left via TRRS cable.
+
+#    define RGBLIGHT_LIMIT_VAL 100
+// #define RGBLIGHT_LAYERS
+
+#    ifndef HSV_BLACK
+#        define HSV_BLACK 0, 0, 0  // Define 'black' color, more like 'LED off' (H, S, V)
+#    endif
+
+// By default, LEDs of the buttons which are pressed turn on.
+// #    define RGBLIGHT_EFFECT_TWINKLE
+
+
+// RGBLIGHT LED MAP
+//     Right 1st row (12 LEDs)
+//     ENCODER
+//     Right 2nd .. 5th row(13, 12, 13, 12 LEDs)
+//     Left 1st .. 5th row (12 LEDs x 5 rows)
+
+ #define RGBLIGHT_LED_MAP { \
```

Comment:
However, it sounds like these are mapped to individual keys? 

If so, you may actually want to use rgb matrix, as that does support ws2812 leds for the matrix type.

Reply:
Yes, my first thought was to use rgb matrix, which I'm familiar with and I like better, instead of using rgb lighting.
The reason I gave up using rgb matrix was that there are duplicate keys on the right hand side.
Some buttons are for same notes. I wanted to turn on both LEDs for them when either of keys are pressed.
Right side have 62 key switches but 24 keys are duplicates. The matrix detects only 62 - 24 = 38 keys.
(There are one rotary encoder so actually 39 keys are detected.)

I'm pretty sure rgb matrix is useful when there are same # of keys and LEDs on the board.
Since I couldn't solve the above task, I gave up using rgb matrix.

If you have better solutions, I'm all ears!
Thanks! 

P.S.
I'll upload the keyboard details how they look and the schematic of the boards later.
I received keycaps today. It looks much like a chromatic button accordion now.

Reply:
Yes, my first thought was to use rgb matrix, which I'm familiar with and I like better, instead of using rgb lighting.
The reason I gave up using rgb matrix was that there are duplicate keys on the right hand side.
Some buttons are for same notes. I wanted to turn on both LEDs for them when either of keys are pressed.
Right side have 62 key switches but 24 keys are duplicates. The matrix detects only 62 - 24 = 38 keys.
(There are one rotary encoder so actually 39 keys are detected.)

For instance, these keys are duplicates. Pressing either of them will output a C note.
![IMG_4440 copy](https://user-images.githubusercontent.com/69518343/91034151-76484f80-e63f-11ea-80aa-d0645c2ba5de.jpg)


I'm pretty sure rgb matrix is useful when there are same # of keys and LEDs on the board.
Since I couldn't solve the above task, I gave up using rgb matrix.

If you have better solutions, I'm all ears!
Thanks! 

P.S.
I upload the keyboard details how they look.

Left hand side (sub)
![IMG_4442](https://user-images.githubusercontent.com/69518343/91024262-a89f8000-e632-11ea-9c5b-e19cb6067b0f.jpg)

Right hand side (main)
![IMG_4440](https://user-images.githubusercontent.com/69518343/91024242-a1787200-e632-11ea-8386-2a31e03aa40e.jpg)

I received keycaps today. It looks much more like a chromatic button accordion now.

Reply:
Ah, that I can understand.

Though... you could create custom animations based on the ones you want, that could handle these edge cases, actually.  https://docs.qmk.fm/#/feature_rgb_matrix?id=custom-rgb-matrix-effects

These would be specific to the keyboard. 

However, this ... would definitely be more work.

Reply:
Yeah, it looks more difficult for me.

Reply:
@drashna, could you "Resolve conversation" if the above is OK?
It's up and running exactly how I want to.
Thank you in advance!